tools/src/benchmarker.cr

```diff
@@ -126,42 +128,64 @@ def frameworks : Array(Target)
   targets
 end
 
-# Benchmark
-# server : server context
-# count  : number of samples
-# threads : number of thread to launch simultaneously
-# requests : numbers of request per thread
-def benchmark(host, count, threads, requests) : BenchResult
-  max : Float64 = -1.0
-  min : Float64 = 100_000.0
-  ave : Float64 = 0.0
-  total : Float64 = 0.0
-
-  count.times do |i|
-    span = client(host, threads, requests)
-    max = span if span > max
-    min = span if span < min
-    total += span
-  end
+class Result
+  JSON.mapping(
+    errors: Int32,
+    latency: {type: Latency, nilable: false},
+    request: {type: Request, nilable: false},
+    percentile: {type: Percentile, nilable: false},
+    throughput: {type: Throughput, nilable: false}
+  )
+end
 
-  ave = total/count.to_f
+class Latency
+  JSON.mapping(
+    average: String,
+    stdev: String,
+    max: String,
+    pmstdev: String,
+  )
+end
 
-  result = BenchResult.new(max, min, ave, total)
+class Request
+  JSON.mapping(
+    average: String,
+    stdev: String,
+    max: String,
+    pmstdev: String,
+    total: Int32,
+    per_second: Float64
+  )
+end
 
-  sleep 5
+class Percentile
+  JSON.mapping(
+    fifty: String,
+    seventy_five: String,
+    ninety: String,
+    ninety_nine: String
+  )
```

Comment:
Showing the 4 or 5 9's percentile stat is also useful and standard as it shows the extreme outliers.